ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/TimestampScheduleBuilder.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.config.GenesisConfigOptions;
+import org.hyperledger.besu.ethereum.chain.BadBlockManager;
+import org.hyperledger.besu.ethereum.core.PrivacyParameters;
+import org.hyperledger.besu.ethereum.privacy.PrivateTransactionValidator;
+import org.hyperledger.besu.evm.internal.EvmConfiguration;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TimestampScheduleBuilder {
```

Comment:
There is a lot of duplication between this class and the ProtocolScheduleBuilder. Is it possible to reuse some of the code from `ProtocolScheduleBuilder`? Or just reuse the `ProtocolScheduleBuilder` passing the timestamp specific pieces?

Reply:
I'll have a think about this. Some of the duplication goes away when we remove the modifiers as part of #4788.

There's enough slight differences that I'm worried it could make it unnecessarily complicated to understand the code.
I don't expect this code is going to change much either.
Also, eventually we'll probably only use the TimestampSchedule for public networks at least and ProtocolSchedule for private use case so they may naturally live separately in the future.

Reply:
OK, I've implemented it here: https://github.com/gezero/besu/pull/1

It was easy enough...question is though, is it worth the extra complexity for the future code reader?

Reply:
After speaking to Jason, decide to add the refactor into this PR.